File: exercise01/Python/solution.py
def ultima_parada(combustivel: int, consumo: int, postos_de_gasolina: list[int]) -> int:
    max_distance = consumo*combustivel
    actual_distance = 0
    for posto_distance in postos_de_gasolina:
        if posto_distance <= max_distance and posto_distance > actual_distance:
            actual_distance = posto_distance
    return actual_distance

File: exercise01/C/solution.c
#include <stdio.h>

void main()
{
    // ENTRADA
    int i, combustivel, consumo, postos_de_gasolina[100];

    printf("Combustivel: ");
    scanf("%d", &combustivel);

    printf("Consumo: ");
    scanf("%d", &consumo);

    printf("Postos de gasolina: ");
    while (scanf("%d", &postos_de_gasolina[i]) == 1)
    {
        i++;
    }

    // PROCESSAMENTO
    int max_distance = consumo * combustivel;
    int actual_distance = 0;
    for (int j = 0; j < i; j++)
    {
        if (postos_de_gasolina[j] <= max_distance && postos_de_gasolina[j] > actual_distance)
        {
            actual_distance = postos_de_gasolina[j];
        }
    }

    // SAIDA
    printf("Ultima parada: %d\n", actual_distance);
}